Server/VikingVault.API/Controllers/CompanyController.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using VikingVault.API.SecurityFilters;
+using VikingVault.DataAccess.Enums;
+using VikingVault.DataAccess.Models;
+using VikingVault.Services;
+using VikingVault.Services.Abstractions;
+
+namespace VikingVault.API.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class CompanyController : ControllerBase
+    {
+        private readonly ICompanyService _companyService;
+
+        public CompanyController(ICompanyService companyService)
+        {
+            _companyService = companyService;
+        }
+        
+        [Authorization(Role=RoleEnum.Admin)]
+        [HttpPost]
+        public ActionResult<User> Post([FromBody] CompanyDTO company)
+        {
+            try
+            {
+                return Ok(_companyService.CreateCompany(company));
+            }
+            catch(CompanyServiceException e)
+            {
+                if(e.Message == "Company already exists")
```

Comment:
It's better to create a more specific exception here and catch that instead of checking for the message. What if you want to change the message of that exception?